paperqa/config.py

```diff
@@ -0,0 +1,413 @@
+import importlib.resources
+import os
+from enum import Enum
+from pathlib import Path
+from typing import ClassVar, assert_never, cast
+
+from pydantic import BaseModel, ConfigDict, Field, field_validator
+from pydantic_settings import (
+    BaseSettings,
+)
+
+from .paths import PAPERQA_DIR
+from .prompts import (
+    citation_prompt,
+    default_system_prompt,
+    qa_prompt,
+    select_paper_prompt,
+    structured_citation_prompt,
+    summary_json_prompt,
+    summary_json_system_prompt,
+    summary_prompt,
+)
+from .types import Answer
+from .utils import hexdigest, pqa_directory
+from .version import __version__
+
+
+class AnswerSettings(BaseModel):
+    evidence_k: int = Field(
+        default=10, description="Number of evidence pieces to retrieve"
+    )
+    evidence_detailed_citations: bool = Field(
+        default=True, description="Whether to include detailed citations in summaries"
+    )
+    evidence_retrieval: bool = Field(
+        default=True,
+        description="Whether to use retrieval instead of processing all docs",
+    )
+    evidence_summary_length: str = Field(
+        default="about 100 words", description="Length of evidence summary"
+    )
+    evidence_skip_summary: bool = Field(
+        default=False, description="Whether to summarization"
+    )
+    answer_max_sources: int = Field(
+        default=5, description="Max number of sources to use for an answer"
+    )
+    answer_length: str = Field(
+        "about 200 words, but can be longer", description="Length of final answer"
+    )
+    max_concurrent_requests: int = Field(
+        default=4, description="Max concurrent requests to LLMs"
+    )
+    answer_filter_extra_background: bool = Field(
+        default=False,
+        description="Whether to cite background information provided by model.",
+    )
+    model_config = ConfigDict(extra="forbid")
```

Comment:
Let's move `model_config` to the top with a newline separating it, to keep Pydantic internals visually separate from the normal fields